pymeasure/instruments/rigol/rigol_ds1000z.py

```diff
@@ -0,0 +1,849 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+try:
+    from enum import StrEnum
+except ImportError:
+    from enum import Enum
+
+    class StrEnum(str, Enum):
+        """Until StrEnum is broadly available from the standard library"""
+        # Python>3.10 remove it.
+
+from typing import Union, Optional
+
+import numpy as np
+
+from pymeasure.instruments import Channel, Instrument, SCPIMixin
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+
+_ANALOG_SOURCES: tuple[str, ...] = ("CHAN1", "CHAN2", "CHAN3", "CHAN4", "MATH")
+_DIGITAL_SOURCES: tuple[str, ...] = tuple(f"D{idx}" for idx in range(16))
+_WAVEFORM_SOURCES: tuple[str, ...] = _ANALOG_SOURCES + _DIGITAL_SOURCES
+_MEMORY_DEPTH_VALUES: tuple[int, ...] = (
+    3000,
+    6000,
+    12000,
+    30000,
+    60000,
+    120000,
+    300000,
+    600000,
+    1200000,
+    3000000,
+    6000000,
+    12000000,
+    24000000,
+)
+_ANALOG_MEMORY_DEPTH_OPTIONS: dict[int, set[int]] = {
+    0: set(_MEMORY_DEPTH_VALUES),
+    1: {12000, 120000, 1200000, 12000000, 24000000},
+    2: {6000, 60000, 600000, 6000000, 12000000},
+    3: {3000, 30000, 300000, 3000000, 6000000},
+    4: {3000, 30000, 300000, 3000000, 6000000},
+}
+_DIGITAL_MEMORY_DEPTH_OPTIONS: dict[int, set[int]] = {
+    0: set(_MEMORY_DEPTH_VALUES),
+    8: {12000, 120000, 1200000, 12000000, 24000000},
+    16: {6000, 60000, 600000, 6000000, 12000000},
+}
+
+
+class AcquireType(StrEnum):
+    """Supported acquisition types for the scope."""
+    # In this mode, the oscilloscope samples the signal at equal time interval to rebuild the
+    # waveform. For most of the waveforms, the best display effect can be obtained using this mode.
+    NORMAL = "NORM"
+    # In this mode, the oscilloscope averages the waveforms from multiple samples to reduce the
+    # random noise of the input signal and improve the vertical resolution. The number of averages
+    # can be set by the `average_count` property. Greater number of averages can lower the noise
+    # and increase the vertical resolution, but will also slow the response of the displayed
+    # waveform to the waveform changes.
+    AVERAGE = "AVER"
+    # In this mode, the oscilloscope acquires the maximum and minimum values of the signal within
+    # the sample interval to get the envelope of the signal or the narrow pulse of the signal
+    # that might be lost. In this mode, signal confusion can be prevented but the noise displayed
+    # would be larger.
+    PEAK = "PEAK"
+    # This mode uses a kind of ultra-sample technique to average the neighboring points of the
+    # sample waveform to reduce the random noise on the input signal and generate much smoother
+    # waveforms on the screen. This is generally used when the sample rate of the digital
+    # converter is higher than the storage rate of the acquisition memory.
```

Comment:
The comments are only visibile if you open this file.

If you add them to the docstring of the enum, it will be visible while using the enum.